great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,390 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional
+
+from great_expectations.core.batch_spec import BatchSpec, PathBatchSpec
+from great_expectations.core.util import AzureUrl, DBFSPath, GCSUrl, S3Url
+from great_expectations.experimental.datasources.data_asset.data_connector.batch_filter import (
+    BatchFilter,
+    build_batch_filter,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.data_connector import (
+    DataConnector,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.regex_parser import (
+    RegExParser,
+)
+
+# TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+# TODO: <Alex>ALEX</Alex>
+# from great_expectations.experimental.data_asset.data_connector.sorter import Sorter
+# TODO: <Alex>ALEX</Alex>
+from great_expectations.experimental.datasources.data_asset.data_connector.util import (
+    batch_definition_matches_batch_request,
+    map_batch_definition_to_data_reference_string_using_regex,
+    map_data_reference_string_to_batch_definition_list_using_regex,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+class DataConnectorStorageDataReferenceResolver:
```

Comment:
This whole class is copied from `great_expectations/datasource/data_connector/file_path_data_connector.py`. Can we reuse this from there?

Reply:
@billdirks It is not copied; it is carefully ported.  There are material differences (in ZEP, `DataAsset` is the only source of the configuration for `DataConnector`).

Reply:
@billdirks This class was deleted (please see `https://github.com/great-expectations/great_expectations/pull/7094#discussion_r1103302404`) based on review/feedback by @donaldheppner -- we utilize one of the key differences between the ZEP and V3 architectures to be able to do this.  Thanks.